pkg/controller/jobframework/reconciler.go

```diff
@@ -135,6 +136,16 @@ func WithEnabledFrameworks(i *configapi.Integrations) Option {
 	}
 }
 
+// WithEnabledExternalFrameworks adds framework names managed by external controller in the Config API.
+func WithEnabledExternalFrameworks(i *configapi.Integrations) Option {
```

Comment:
Maybe just pass the string slice of external integrations.

Reply:
In my opinion the nil check should be performed by the caller, also the fact that we get that pointer when the object is created and only use it when the option is applied can be seen as problematic.

In case config.Integratons is nil, we are failing before this because of:
https://github.com/kubernetes-sigs/kueue/blob/6cceaa070f1d310d9e7da817f66c0f7f31e2623b/cmd/kueue/main.go#L272

For this PR, I guess, we can have it like this as well but we'll need a cleanup PR for this and `WithEnabledFrameworks` .

Reply:
Uhm, that sounds reasonable.
Let me try to refactor the `WithEnabledFrameworks` and `WithIntegrationOptions` in another PR, first.
Thanks.

/hold

Reply:
Wait, I just remembered the reason why I didn't implement the null pointer check here.

https://github.com/kubernetes-sigs/kueue/blob/6cceaa070f1d310d9e7da817f66c0f7f31e2623b/cmd/kueue/main.go#L272

We already checked if the `cfg.integrations` is not null here: https://github.com/kubernetes-sigs/kueue/blob/6cceaa070f1d310d9e7da817f66c0f7f31e2623b/pkg/config/validation.go#L128-L130

So, I don't this that we should verify the null pointer for the `WithIntegrationOptions`.
But I agree with passing the `integrations.Frameworks` to the `WithEnabledFrameworks` instead of the `integrations`.

I just can change the arg of the `WithEnabledFrameworks` in this PR.

Reply:
/hold cancel

Reply:
> I just can change the arg of the WithEnabledFrameworks in this PR.

It seems that we need to change many places if we want to change the arg of the `WithEnabledFrameworks`.
So, let me do it in the follow-ups.

Reply:
/hold cancel

Reply:
DONE.